### PR TITLE
Allow to specify a _target_ value for the monitored quantity in EarlyStopping callback

### PR DESCRIPTION
**Pull request by giuscri**
> As of 354bd35, EarlyStopping callback allows only to specify a baseline. That value is used to stop training when monitored quantity is worse than the baseline. To my understanding, what's missing is the ability to specify a value to reach for the monitored quantity before considering to stop training. Say you want to stop training if stable but you want to reach at least an accuracy of 95%. Right now you can't.

https://github.com/keras-team/keras/issues/10586

### CODE REVIEW

File: tests/keras/test_callbacks.py [hunk 1]
```diff
@@ -287,32 +287,101 @@ def __init__(self):
...
+    early_stop.on_train_begin()
+
+    for epoch in range(len(accuracies)):
+        epochs_trained += 1
+        early_stop.on_epoch_end(epoch, logs={'val_acc': accuracies[epoch]})
+
+        if early_stop.model.stop_training:
+            break
+
+    assert epochs_trained == 3
```
Comment (taehoonlee) [inline tests/keras/test_callbacks.py]: Thank you for the PR, @giuscri. Why is `epochs_trained` 3? The accuracy has been already reached in the second epoch.

**Comment by giuscri** [comment]
`reach` is a lower bound for the accuracy. An accuracy of at least 90% is requested by the user but in general the accuracy reached is higher.

Model is trained for three epochs: the first two are needed for the model to reach the specified accuracy. The third epochs is done to check there's no improvement.

**Comment by fchollet** [comment]
You can very easily implement this on your own by subclassing the `EarlyStopping` callback, e.g.

````python

class MyEarlyStopping(EarlyStopping):

  def __init__(self, minimum_target=None, **kwargs):
    self.minimum_target = minimum_target
    super(MyEarlyStopping, self).__init__(**kwargs)

  def on_epoch_end(self, epoch, logs):
    current = logs.get(self.monitor)
    if self.minimum_target and if self.monitor_op(self.minimum_target, current):
          super(MyEarlyStopping, self).on_epoch_end(epoch, logs)
```

Every feature has a cost that goes beyond the initial PR: maintenance cost, documentation cost, and cognitive cost for users. In general, we would rather not add new keyword arguments to support various individual niche use cases. In such cases, it is better for users to extend the current functionality to match their own needs (e.g. via subclassing).

**Comment by Attila94** [comment]
Thanks for the snippet @fchollet, works perfectly after a minor correction:

`class MyEarlyStopping(EarlyStopping):
  def __init__(self, minimum_target=None, **kwargs):
    self.minimum_target = minimum_target
    super(MyEarlyStopping, self).__init__(**kwargs)
  def on_epoch_end(self, epoch, logs):
    current = logs.get(self.monitor)
    if self.minimum_target and not self.monitor_op(self.minimum_target, current):
          super(MyEarlyStopping, self).on_epoch_end(epoch, logs)`
